src/main/java/com/google/codeu/data/Message.java

```diff
@@ -33,15 +33,22 @@
    * Generates a random ID and uses the current system time for the creation time.
    */
   public Message(String user, String text, String recipient) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient);
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), recipient, null);
   }
 
   public Message(UUID id, String user, String text, long timestamp, String recipient) {
+    this(id, user, text, timestamp, recipient, null);
+  }
+
+  /** Constructs a new Message written by the user, if they uploaded an image. */
+  public Message(UUID id, String user, String text, long timestamp, String recipient,
+      String imageUrl) {
     this.id = id;
     this.user = user;
     this.text = text;
-    this.timestamp = timestamp;
+    this.timestamp = timestamp; 
```

Comment:
It looks like a random space got added to the end of this line.

Reply:
!! Nice catch! Thanks!